pkg/services/grpcserver/service.go

```diff
@@ -74,6 +75,7 @@ func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, authe
 	// services which implement ServiceAuthFuncOverride interface.
 	// See https://github.com/grpc-ecosystem/go-grpc-middleware/blob/main/interceptors/auth/auth.go#L30.
 	opts = append(opts, []grpc.ServerOption{
+		grpc.StatsHandler(otelgrpc.NewServerHandler()),
 		grpc.ChainUnaryInterceptor(
 			grpcAuth.UnaryServerInterceptor(authenticator.Authenticate),
 			interceptors.TracingUnaryInterceptor(tracer),
```

Comment:
should the old one be removed?

Reply:
Tried things out without it and traces are still getting propagated, so yep, guess we don't need it.

I'm not sure about the streaming tracing interceptors though.